crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1104,10 +1114,20 @@ where
                             //    * this case represents a potentially long range of blocks to
                             //      download and execute
 
-                            // TODO: use a range request
-                            // TODO: add method for the range count, and integrate that into the
-                            // upcoming `exceeds_pipeline_run_threshold` method
-                            self.sync.download_full_block(missing_parent.hash);
+                            if let Some(distance) = self
+                                .distance_from_local_tip(canonical_tip_num, missing_parent.number)
+                            {
+                                self.sync.download_block_range(missing_parent.hash, distance)
+                            } else {
+                                // This means the local tip is greater than the missing
+                                // parent?
+                                // TODO: when can this happen?
+                                //
+                                // is this distance_from_local_tip method just overengineering?
+                                //
+                                // can we convert this into a range download?
+                                self.sync.download_full_block(missing_parent.hash);
```

Comment:
I think this would only be possible for outdated side chains where the disconnected block could potentially attach to some parent canonical block (not the current tip).

I think we should just start downloading this as we do now, perhaps this could happen when we're having a reorg of some kind, downloading single blocks seems fine